src/Illuminate/Testing/TestResponse.php

```diff
@@ -1739,6 +1740,8 @@ public function __call($method, $args)
             return $this->macroCall($method, $args);
         }
 
+        throw_if(!$this->baseResponse instanceof Response);
```

Comment:
This is wrong, you would want to allow access for methods in `Symfony\Component\HttpFoundation\Response` as well including` isSuccessful()` and `isRedirection()` which is used directly within this class.

Reply:
By using the @template annotation, we can dynamically determine the response class, allowing for better static analysis and type safety. This approach ensures that the __call method can handle different response types without causing errors in PHPStan or similar tools.